lib/persistence/repositories/impl/account_repository_impl.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:coffeecard/model/login.dart';
+import 'package:coffeecard/persistence/http/coffee_card_api_client.dart';
+import 'package:coffeecard/persistence/repositories/account_repository.dart';
+import 'package:coffeecard/persistence/storage/secure_storage.dart';
+import 'package:dio/dio.dart';
+import 'package:logger/logger.dart';
+
+class AccountRepositoryImpl implements AccountRepository {
+  final CoffeeCardApiClient _restClient;
+  final Logger _logger;
+  final SecureStorage _storage;
+
+  AccountRepositoryImpl(this._restClient, this._logger, this._storage);
+
+  @override
+  Future login(String userName, String password) async {
+    var login = Login(userName, password, "");
+
+    try {
+      var token = await _restClient.login(login);
+      await _storage.saveToken(token.token);
+    } on DioError catch (error) {
+      final httpResponse = error.response;
+      _logger.e("API Error ${httpResponse.statusCode} ${httpResponse.statusMessage}");
+      rethrow;
```

Comment:
A proper way of handling rest errors will be addressed in https://trello.com/c/ZV5Ul2N4/33-error-handling-in-repositories